capi-configuration/README.md

```diff
@@ -0,0 +1,333 @@
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1alpha3
+          kind: KubeadmConfig
+          name: capa-mp-0
+      clusterName: capa
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+        kind: AWSMachinePool
+        name: capa-mp-0
+      version: v1.16.8
+```
+From this custom resource we can now extract the values which are static and configurable.
+So values which a customer might want to change the default of _and_ which we can default statically independent of any other content in the custom resource.
+
+In this particular case there are two values which are both static and configurable:
+- `spec.replicas`
+- `spec.template.spec.version`
+
+We can now construct a `ConfigMap` with these values which are then going to be used by any defaulting tool/webhook:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: machinepool-21-0-0
+  namespace: giantswarm
+data:
+  spec:
+    replicas: 3
+    template:
+      spec:
+        version: v1.21.4
+```
+It is important to note that a assumption is that **no** values in the base file are immutable!
+
+### Cluster-api context dependent values
+
+We have some cases in `cluster-api` where a value in a custom resource is neither part of the fundamental structure nor is it easy to statically default.
+
+These values are usually highly dependent on the context in which the cluster is created in.
+For example the `CIDR` in which the cluster is created.
+We currently assume that these values will be defaulted by custom webhooks or operators with significant provider specific domain knowledge.
+
+### Putting together a Giant Swarm release
+
+A Giant Swarm releases is just a set of `cluster-api` base files as defined previously with added `apps`.
+
+The following folder structure would be contained in `giantswarm/example-customer`:
+```yaml
+aws:
+  v21.0.0:
+    base:
+      cluster.yaml # configmap containing the base file (defined above)
+      aws-cluster.yaml
+      machinepool.yaml
+      awsmachinepool.yaml
+      ...
+      net-exporter.yaml # configmap containing version information for defaulting!
+    overlay:
+      special-overlay.yaml # Added by customer/AE/SA to e.g. overlay a different flatcar version!
+  v22.0.0:
+    base:
+      ...
+    overlay:
+      ...
+```
+
+In the management cluster a release is then simply a set of configmaps.
+The configmaps for e.g. release `v21.0.0` will be applied through argoCD by taking the `base` configmaps, applying any `overlay` from a customer and then deploying them.
+
+As shown in previous examples these configmaps would contain the release in their name (e.g. `machinepool-21-0-0`).
+Any in-cluster webhook or operator can then easily navigate to these configmaps to look up defaults for a cluster.
+
+A cluster creation workflow can now look as follows:
+1. Client (e.g. `happa`) submits `--dry-run` with `fundamental` custom resources to management cluster.
```

Comment:
I think the important part is that they send the minimal CR, e.g. from the example above?
```suggestion
1. Client (e.g. `happa`) submits `--dry-run` with `minimal` custom resources to management cluster.
```